bin/run_clang_tidy.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -e
+
+# TODO
+# Actually apply changes (add -fix flag)
+BUILD_PATH=/build/faasm/
+CONFIG=${FAASM_ROOT}/.clang-tidy
+
+if [ -z "$1" ]; then
+    TARGET_DIR="."
+else
+    TARGET_DIR=$1
+fi
+
+pushd ${TARGET_DIR} >> /dev/null
+
+FILES=$(git ls-files {libs,func}/{"*.h","*.cpp","*.c"})
```

Comment:
As discussed offline, we only run the linting on `libs` and `func`.

Reply:
Why not `src` and `test`? Sorry I can't remember. Just because it spits out too many errors?

Reply:
No. It's just for some reason I had noted down to run them only un `func` and `libs`. But it was something we said offline, discussing out loud, so no problem in adding these.